modules/custom-tuning-specification.adoc

```diff
@@ -43,43 +44,77 @@ profile:
+The `recommend:` section is a list of items to recommend the profiles based on
+a selection criteria.
 
+[source, yaml]
 ----
 recommend:
-- match:                              # optional; if omitted, profile match is assumed unless a profile with a higher matches first
-  <match>                             # an optional array
-  priority: <priority>                # profile ordering priority, lower numbers mean higher priority (0 is the highest priority)
-  profile: <tuned_profile_name>       # e.g. tuned_profile_1
-
+<recommend-item-1>
 # ...
-
-- match:
-  <match>
-  priority: <priority>
-  profile: <tuned_profile_name>       # e.g. tuned_profile_n
+<recommend-item-n>
 ----
 
-If `<match>` is omitted, a profile match (for example, `true`) is assumed.
+The individual items of the list:
+
+[source, yaml]
+----
+- machineConfigLabels: <1>
+    <mcLabels> <2>
+  match: <3>
+  <match> <4>
+  priority: <priority> <5>
+  profile: <tuned_profile_name> <6>
+----
+<1> Optional.
+<2> A dictionary of key/value MachineConfig labels. The keys must be unique.
+<3> If omitted, profile match is assumed unless a profile with a higher priority matches first or `machineConfigLabels` is set.
+<4> An optional list.
+<5> Profile ordering priority. Lower numbers mean higher priority (0 is the highest priority).
+<6> A tuned profile to apply on a match. For example `tuned_profile_1`.
 
-`<match>` is an optional array recursively defined as follows:
+`<match>` is an optional list recursively defined as follows:
 
+[source, yaml]
 ----
-- label: <label_name>     # node or pod label name
-  value: <label_value>    # optional node or pod label value; if omitted, the presence of <label_name> is enough to match
-  type: <label_type>      # optional node or pod type ("node" or "pod"); if omitted, "node" is assumed
-  <match>                 # an optional <match> array
+- label: <label_name> <1>
+  value: <label_value> <2>
+  type: <label_type> <3>
+  <match> <4>
 ----
+<1> Node or pod label name.
+<2> Optional node or pod label value. If omitted, the presence of `<label_name>` is enough to match.
+<3> Optional node or pod type ("node" or "pod"). If omitted, "node" is assumed.
+<4> An optional `<match>` list.
 
 If `<match>` is not omitted, all nested `<match>` sections must also evaluate to
 `true`. Otherwise, `false` is assumed and the profile with the respective
 `<match>` section will not be applied or recommended. Therefore, the nesting
 (child `<match>` sections) works as logical AND operator. Conversely, if any
-item of the `<match>` array matches, the entire `<match>` array evaluates to
-`true`. Therefore, the array acts as logical OR operator.
-
-.Example
-
+item of the `<match>` list matches, the entire `<match>` list evaluates to
+`true`. Therefore, the list acts as logical OR operator.
+
+If `machineConfigLabels` is defined, MachineConfigPool based matching is turned on
+for the given `recommend:` list item. This is in addition (logical OR) to the matching
```

Comment:
s/This is in addition to/Logical OR is in addition to 
(I think that's what is meant here with the parentheticals)

Reply:
So this obviously needs to be explained better and I wonder if you can help here.  I had a go at this, comments welcome.

Reply:
@jmencak I made a suggestion. Thanks!